### PR TITLE
mesh_demo 

### PR DESCRIPTION
**Pull request by vikrant8052**
This is a simple application demonstrating a nRF52840_PDK based #BluetoothMesh node.

Root element has following Models:
1. Generic OnOff Server
2. Generic OnOff Client
3. Generic Level Server
4. Generic Level Client

We could provision & configure it using following mentioned options:
1. BlueZ #meshctl daemon
2. #nRFMesh iOS App
3. Silicon Labs Bluetooth Mesh App

### CODE REVIEW

File: samples/boards/nrf52/mesh/onoff_level_app/README.rst [hunk 1]
```diff
@@ -0,0 +1,101 @@
+.. _bluetooth-mesh-onoff-sample:
```
Comment (dbkinder) [inline samples/boards/nrf52/mesh/onoff_level_app/README.rst]: These document labels must be unique across all the zephyr documents and this name is already used by samples/boards/nrf52/mesh/onoff-app/README.rst  (I suspect you copied that one to create this doc).  Change this document label to maybe

    .. _bluetooth-mesh-onoff-level-sample:

File: samples/boards/nrf52/mesh/onoff_level_app/README.rst [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+Bluetooth: Mesh OnOff & Level Model
+###########################
+
+Overview
+********
+
+This is a simple application demonstrating a Bluetooth mesh node.
+
+Root element has Generic OnOff Server, Generic OnOff Client, Generic Level Server, Generic Level Client models.
```
Comment (dbkinder) [inline samples/boards/nrf52/mesh/onoff_level_app/README.rst]: Start the sentence with:

    The node's root element has ...

and add an ``and`` after the last comma in the sentence:

    ... Generic Level Server, and Generic Level Client models.

File: samples/boards/nrf52/mesh/onoff_level_app/README.rst [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+Prior to provisioning, an unprovisioned beacon is broadcast that contains
+a unique UUID. It is obtained from the device address set by Nordic in the
+FICR. Each button controls the state of its
```
Comment (dbkinder) [inline samples/boards/nrf52/mesh/onoff_level_app/README.rst]: for clarity, add:

   in the Factory information configuration registers (FICR).

File: samples/boards/nrf52/mesh/onoff_level_app/README.rst [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+corresponding LED and does not initiate any mesh activity.
+
+Association of Models with h/w (in case of nRF52840-PDK board)
```
Comment (dbkinder) [inline samples/boards/nrf52/mesh/onoff_level_app/README.rst]: Recommend spelling out ``hardware`` and rewrite this section as:

    Associations of Models with hardware
    ************************************

    For the nRF52840-PDK board, these are the model associations:

    * LED1 is associated with generic OnOff Server
    * Button1 and Button2 are associated with generic OnOff Client: 

      * [Button1 : ON]
      * [Button2: OFF]
    * LED3 is associated with generic Level Server:

      * [if (Level < 50) LED3: OFF else LED3: ON]
    * Button3 and Button4 are associated with generic Level Client: 

      * [Button3: publishes Level = 25]
      * [Button4: publishes Level = 100]

Note the blank lines and indentation is important.

File: samples/boards/nrf52/mesh/onoff_level_app/README.rst [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+LED1 is associated with Gen. OnOFF Server
+
+Button1 & Button2 are associated with Gen. OnOff Client [Button1 : ON & Button 2: OFF]
+
+LED3 is associated with Gen. Level Server [if (Level < 50) LED3: OFF ...if (Level >= 50) LED3: ON]
+
+Button3 & Button4 are associated Gen. Level Client [Button3: publishes Level = 25 .....Button4: publishes Level = 100]
+
+-------------------------------------------------------------------------------------------------------------------------
```
Comment (dbkinder) [inline samples/boards/nrf52/mesh/onoff_level_app/README.rst]: delete this line

File: samples/boards/nrf52/mesh/onoff_level_app/README.rst [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+Requirements
+************
+
+This sample has been tested on the Nordic nRF52840-PDK board, but would
+likely also run on the nrf52_pca10040 board.
+
+Building and Running
+********************
+
+This sample can be found under :file:`samples/boards/nrf52/mesh/onoff-app` in the
```
Comment (dbkinder) [inline samples/boards/nrf52/mesh/onoff_level_app/README.rst]: should be ``onoff_level_app``

File: samples/boards/nrf52/mesh/onoff_level_app/README.rst [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+Zephyr tree.
+
+The following commands build the application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/boards/nrf52/mesh/onoff-app
```
Comment (dbkinder) [inline samples/boards/nrf52/mesh/onoff_level_app/README.rst]: same here, ``onoff_level_app``

File: samples/boards/nrf52/mesh/onoff_level_app/README.rst [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+   :board: nrf52840_pca10056
+   :goals: build flash
+   :compact:
+
+Prior to provisioning, each button controls its corresponding LED as one
+would expect with an actual switch.
+
+Provisioning is done using the BlueZ meshctl utility. Below is an example that
+binds button 1&2 and LED 1 to application key 1. It then configures button 1&2
```
Comment (dbkinder) [inline samples/boards/nrf52/mesh/onoff_level_app/README.rst]: You called these ``Button1`` and ``Button2`` previously (without a space), so I'd be consistent here and write this as:

    Here is an example that binds Button1, Button2, and LED1 to
    application key 1. It then configures Button1 and Button2
    to publish to group 0xc000 and LED1 to subscribe to that group.

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by vikrant8051** [comment]
@dbkinder I've make suggested changes.

**Pull request closed without merging**